nixjs-rt/src/lib.ts

```diff
@@ -227,15 +245,33 @@ export abstract class NixType {
    */
   abstract typeOf(): string;
 
+  /**
+   * Returns a human-readable string representation of this value, that can be inserted into a sentence.
+   *
+   * For example, "a string", "an array", etc.
+   *
+   * Static functions can't be made abstract, so abstract is omitted here.
+   */
+  static toHumanReadable(): string {
```

Comment:
This function has a couple of abstraction mismatches:
- it's a static function but it talks about the instance
- the name suggests it would convert the the type to a readable representation.

How about not adding a new function to `NixType` but instead solve it in the error message-building API? For example, say you have an error message like this:
```typescript
err`Expected ${bold(NixList)}, but got ${boldValueType(value)}`
```
Then the `boldValueType` function could do the conversion of the type of the value to a string like:
```typescript
err`an ${bold("attrset")}`
```

Reply:
I'm not really sure what you're saying. So `bold` would have an if statement chain to check what instance it is? And then return accordingly? Instead of just using polymorphism.

I understand the arguments for both, but I'm not sure how you'd do this without an if statement chain